aws/resource_aws_cloudfront_distribution_test.go

```diff
@@ -44,6 +44,31 @@ func TestAccAWSCloudFrontDistribution_S3Origin(t *testing.T) {
 	})
 }
 
+func TestAccAWSCloudFrontDistribution_S3OriginNoRestrictions(t *testing.T) {
```

Comment:
Since we're changing the attribute from required to optional, I think it would make more sense to actually remove it from all the other test configurations and just ensure we have "`_restrictions`" test(s) that do actually try to set it.